nucypher/primitives.py

```diff
@@ -0,0 +1,66 @@
+class VersionedBytes:
+
+    class NucypherNeedsUpdateException(BaseException):
+        """This node cannot instantiate a class from data created by a newer version of NuCypher."""
+
+    def __new__(cls, *args, **kwargs):
+        """
+        When instantiating a specific version of a versioned Bytestring,
+        we want to automatically instantiate the latest version without
+        requiring implementors to have to deal with any of this.
+
+        If we are instantiating a specifically versioned class, allow for that also
+        """
+
+        if not hasattr(cls, 'version'):
+            versioned_subclasses = VersionedBytes.__get_versioned_subclasses(cls)
+            # take the highest
+            cls = sorted(versioned_subclasses, key=lambda x: x.version)[-1]
+        return super(VersionedBytes, cls).__new__(cls)
+
+    @classmethod
+    def __get_versioned_subclasses(cls, klass):
+        return [cls for cls in klass.__subclasses__() if hasattr(cls, 'version')]
```

Comment:
If I understand correctly - this is an effort to allow for continued backward compatibility of versioned entities?

Reply:
Yep.